api/v1/numaresourcesoperator_types.go

```diff
@@ -226,15 +226,25 @@ func init() {
 
 func (ngc *NodeGroupConfig) ToString() string {
 	if ngc == nil {
-		return ""
+		return "nil"
```

Comment:
let's see what's the string representation of `nil`

Reply:
https://go.dev/play/p/Jx-o4sG1-dy

Reply:
I followed the k8s convention as in https://github.com/kubernetes/apimachinery/blob/v0.31.2/pkg/apis/meta/v1/generated.pb.go#L4387 (and other String() methods) 
On another hand, I think we can use stringbuilder or similar to construct the string instead of sprintf for such long string presentation.

Reply:
ok, this is better. No need to change

Reply:
The reason for this is that when ToString is called with empty values,some of these (like the v1.Labelselector presenting the MCP selector of the node group) is shown as "nil" while the internal ones lie ng.Config empty value is "" which doesn't look good and may confuse the readers (us mostly).